compiler/rustc_data_structures/src/dyn_send_sync_impl.rs

```diff
@@ -39,15 +39,37 @@ cfg_if!(
 
         macro_rules! impls_dyn_send {
             ($([$t2: ty $(where $($generics2: tt)*)?])*) => {
-                $(unsafe impl$(<$($generics2)*>)? DynSend for $t2 where $t2: Send{})*
```

Comment:
Was the `$t2: Send` bound causing problems? It seemed like a useful safeguard. Could you split impls which work with it out to a separate macro?

Reply:
OK. I split them to two macroes